can/interfaces/socketcan/socketcan.py

```diff
@@ -473,12 +505,39 @@ def send(self, msg, timeout=None):
             if not ready_send_sockets:
                 raise can.CanError("Timeout while sending")
 
+        data = build_can_frame(msg)
         try:
-            self.socket.sendall(build_can_frame(msg))
+            if self.channel == "" and HAS_NATIVE_SUPPORT and msg.channel:
+                self.socket.sendto(data, (msg.channel, ))
```

Comment:
> Return the number of bytes sent.

I guess we should make sure the rest gets sent as well (in a loop or handled completely in a auxiliary method) if the returned number of bytes is less than `len(data)`.